apps/client/src/common/components/multi-part-progress-bar/MultiPartProgressBar.tsx

```diff
@@ -0,0 +1,53 @@
+import { clamp } from '../../utils/math';
+
+import './MultiPartProgressBar.scss';
+
+interface MultiPartProgressBar {
+  now?: number;
+  complete?: number;
+  normalColor?: string;
+  warning?: number;
+  warningColor?: string;
+  danger?: number;
+  dangerColor?: string;
+  hidden?: boolean;
+  className?: string;
+  height?: string;
```

Comment:
these are all values used in the views.

Reply:
I guess I should have been more specific.
What I mean is that it shouldn't be all undefined

Some of this might be wrong, but take the gist of it...
- we will always have a now, sometimes is null
- we will always have a complete, sometimes is null
- ... same for warning and danger, potentially for the colours


The only true optional fields should be hidden, classname and height

Reply:
ahh got ya ..... this is basically taken from your progress bar 😉 apart from the options i added of course

Reply:
Hahaha. I have written a fair share eod questionable code. Depends on the day 😅